cpp/include/cudf/detail/reduction_operators.cuh

```diff
@@ -95,6 +95,23 @@ struct simple_op {
     return element_transformer{};
   }
 
+  /**
+   * @brief Get transformer functor for transforming input column pair iterator
+   * which inturn is used by reduction binary operator
+   *
+   * @tparam ResultType output type for element transformer
+   *
+   * @return element transformer functor object
+   */
+  template <typename ResultType,
+            typename std::enable_if_t<is_template_of<thrust::pair, ResultType>::value>* = nullptr>
+  auto get_null_replacing_element_transformer()
+  {
+    using element_transformer = typename Derived::transformer<typename ResultType::first_type>;
+    return null_replacing_transformer<typename ResultType::first_type, element_transformer>{
+      get_identity<typename ResultType::first_type>(), element_transformer{}};
+  }
```

Comment:
I think it would be clearer to remove the `is_template_of` bound and just use the real result type here instead of `thrust::pair`
```suggestion
  template <typename ResultType>
  auto get_null_replacing_element_transformer()
  {
    using element_transformer = typename Derived::transformer<ResultType>;
    return null_replacing_transformer<ResultType, element_transformer>{
      get_identity<ResultType>(), element_transformer{}};
  }
```